### PR TITLE
docs: add a brief introduction in README

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Resolves: #58

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,10 @@
 <a href="https://oras.land/"><img src="https://oras.land/img/oras.svg" alt="banner" width="100px"></a>
 </p>
 
+`oras-credentials-go` is a credential management library designed for [`oras-go`](https://github.com/oras-project/oras-go). It supports reading, saving, and removing credentials from Docker configuration files and external credential stores that follow the [Docker credential helper protocol](https://docs.docker.com/engine/reference/commandline/login/#credential-helper-protocol), while not handling credential encryption and decryption.
```
Comment (TerryHowe) [inline README.md:14]:
```suggestion
`oras-credentials-go` is a credential management library designed for [`oras-go`](https://github.com/oras-project/oras-go). It supports reading, saving, and removing credentials from Docker configuration files and external credential stores that follow the [Docker credential helper protocol](https://docs.docker.com/engine/reference/commandline/login/#credential-helper-protocol). It does not handle credential encryption and decryption. 
```

Comment (Wwwsylvia) [reply README.md]: @TerryHowe Thanks for the suggestion! Updated.

**Comment by Wwwsylvia** [comment]
Wondering why this PR triggers test coverage... 🤨

File: README.md [hunk 2]
```diff
@@ -11,6 +11,10 @@
 <a href="https://oras.land/"><img src="https://oras.land/img/oras.svg" alt="banner" width="100px"></a>
 </p>
 
+`oras-credentials-go` is a credential management library designed for [`oras-go`](https://github.com/oras-project/oras-go). It supports reading, saving, and removing credentials from Docker configuration files and external credential stores that follow the [Docker credential helper protocol](https://docs.docker.com/engine/reference/commandline/login/#credential-helper-protocol). It does not handle credential encryption and decryption.
+
+Once stable, this library will be merged into `oras-go`. (See [discussion](https://github.com/oras-project/oras-credentials-go/discussions/80))
```
Comment (FeynmanZhou) [inline README.md:16]:
```suggestion
Once it has a fairly stable version, this library will be merged into `oras-go`. (See [discussion](https://github.com/oras-project/oras-credentials-go/discussions/80))
```

Comment (FeynmanZhou) [reply README.md:16]: In general, `Stable` means v1.0.0 but oras-credential-go will only cut a pre-release (v1.0.0-rc.1)

Comment (Wwwsylvia) [reply README.md:16]: How about 
> Once it reaches a fairly stable version (e.g. `v1.0.0-rc.1`), this library will be merged into `oras-go`?

File: README.md [hunk 3]
```diff
@@ -11,6 +11,10 @@
 <a href="https://oras.land/"><img src="https://oras.land/img/oras.svg" alt="banner" width="100px"></a>
 </p>
 
+`oras-credentials-go` is a credential management library designed for [`oras-go`](https://github.com/oras-project/oras-go). It supports reading, saving, and removing credentials from Docker configuration files and external credential stores that follow the [Docker credential helper protocol](https://docs.docker.com/engine/reference/commandline/login/#credential-helper-protocol). It does not handle credential encryption and decryption.
```
Comment (shizhMSFT) [inline README.md:14]: The sentence
> It does not handle credential encryption and decryption

is bit confusing. Rewording requested.

File: README.md (hunk 2 above)
Comment (shizhMSFT) [inline README.md:16]: and... later, this repo will be archived.

File: README.md (hunk 3 above)
Comment (Wwwsylvia) [reply README.md:14]: Could you elaborate a bit on the confusing part?

Comment (Wwwsylvia) [reply README.md]: Let's just remove this sentence then.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**